### DISCUSSION TITLE
[5.1.4][p.291] ApiService에서 catch & redirect

### DISCUSSION THREAD

**Discussion by OverRaddit** [fsoftwareengineer/todo-application#93] [질문과 답변] [answered]
<img width="422" alt="스크린샷 2022-03-04 오후 10 38 54" src="https://user-images.githubusercontent.com/30787477/156773386-ffc227b0-0a3e-4109-b1f2-93a1cf9386ac.png">
실습코드 5-5를 그대로 옮겼는데도

http://localhost:3000/ 로 접속 시
http://localhost:3000/login 으로 redirect되지 않는 현상이 발생했습니다.

책과 깃허브의 코드가 다른 부분이 있어
깃허브를 기준으로 수정한 상태였습니다.
return Promise.reject(json) => return Promise.reject(error)

error.status의 값을 조사하기 위해 console.log를 찍어본 결과
response.json()에서 response가 json의 형식이 아니므로 
json으로 변환을 할때 문제가 발생함을 깨닫고 json으로 변환하는 부분을 삭제하였습니다.

<img width="694" alt="스크린샷 2022-03-04 오후 10 42 35" src="https://user-images.githubusercontent.com/30787477/156773912-999702de-73d9-476d-8fc8-12b1033bc59b.png">
json변환 부분을 제거하는 것으로 문제는 사라지고
정상적으로 redirect시키는데 성공하였습니다.

다만 아직 왜 json으로 변환에 오류가 생기는 지
이해하지 못하여 도움을 구하고자 질문드리게 되었습니다!

**Comment by OverRaddit** [comment, marked as answer]
https://github.com/fsoftwareengineer/todo-application/discussions/30

이미 다른 분께서 질문해주신 내용이어서 해결하였습니다!

.json을 제거하니 cors오류가 발생하여 진행이 어려운 상황이었고 링크한 내용대로
스프링 버전을 낮추니 해결되었습니다.

**Reply by melonicedlatte** [reply]
저도 도움 받았습니다! 감사합니다!
